diff --git a/tensorflow/tsl/mkl/build_defs.bzl b/tensorflow/tsl/mkl/build_defs.bzl
index eaa0b2dbde7..9d709f8abf5 100644
--- a/tensorflow/tsl/mkl/build_defs.bzl
+++ b/tensorflow/tsl/mkl/build_defs.bzl
@@ -33,8 +33,9 @@ def if_mkl(if_true, if_false = []):
     """
     return select({
         "@org_tensorflow//tensorflow/tsl/mkl:build_with_mkl_aarch64": if_true,
-        "@org_tensorflow//tensorflow/tsl:linux_x86_64": if_true,
-        "@org_tensorflow//tensorflow/tsl:windows": if_true,
+        "@org_tensorflow//tensorflow/tsl/mkl:build_with_mkl_lnx_x64": if_true,
+        "@org_tensorflow//tensorflow/tsl/mkl:build_with_mkl_lnx_openmp": if_true,
+        "@org_tensorflow//tensorflow/tsl/mkl:build_with_mkl_windows_openmp": if_true,
         "//conditions:default": if_false,
     })

@@ -102,9 +103,9 @@ def mkl_deps():
     """
     return select({
         "@org_tensorflow//tensorflow/tsl/mkl:build_with_mkl_aarch64": ["@mkl_dnn_acl_compatible//:mkl_dnn_acl"],
-        "@org_tensorflow//tensorflow/tsl:linux_x86_64_with_onednn_v2": ["@mkl_dnn_v1//:mkl_dnn"],
-        "@org_tensorflow//tensorflow/tsl:linux_x86_64_with_onednn_v3": ["@onednn_v3//:mkl_dnn"],
-        "@org_tensorflow//tensorflow/tsl:windows": ["@mkl_dnn_v1//:mkl_dnn"],
+        "@org_tensorflow//tensorflow/tsl/mkl:build_with_mkl_lnx_x64":  ["@mkl_dnn_v1//:mkl_dnn"],
+        "@org_tensorflow//tensorflow/tsl/mkl:build_with_mkl_lnx_openmp":  ["@mkl_dnn_v1//:mkl_dnn"],
+        "@org_tensorflow//tensorflow/tsl/mkl:build_with_mkl_windows_openmp": ["@mkl_dnn_v1//:mkl_dnn"],
         "//conditions:default": [],
     })

