diff --git a/third_party/mkl/build_defs.bzl b/third_party/mkl/build_defs.bzl
index 806b157bad6..6c9f0172b97 100644
--- a/third_party/mkl/build_defs.bzl
+++ b/third_party/mkl/build_defs.bzl
@@ -33,8 +33,9 @@ def if_mkl(if_true, if_false = []):
     """
     return select({
         "@org_tensorflow//third_party/mkl:build_with_mkl_aarch64": if_true,
-        "@org_tensorflow//tensorflow:linux_x86_64": if_true,
-        "@org_tensorflow//tensorflow:windows": if_true,
+        "@org_tensorflow//third_party/mkl:build_with_mkl_lnx_x64": if_true,
+        "@org_tensorflow//third_party/mkl:build_with_mkl_lnx_openmp": if_true,
+        "@org_tensorflow//third_party/mkl:build_with_mkl_windows_openmp": if_true,
         "//conditions:default": if_false,
     })

@@ -102,8 +103,9 @@ def mkl_deps():
     """
     return select({
         "@org_tensorflow//third_party/mkl:build_with_mkl_aarch64": ["@mkl_dnn_acl_compatible//:mkl_dnn_acl"],
-        "@org_tensorflow//tensorflow:linux_x86_64": ["@mkl_dnn_v1//:mkl_dnn"],
-        "@org_tensorflow//tensorflow:windows": ["@mkl_dnn_v1//:mkl_dnn"],
+        "@org_tensorflow//third_party/mkl:build_with_mkl_lnx_x64":  ["@mkl_dnn_v1//:mkl_dnn"],
+        "@org_tensorflow//third_party/mkl:build_with_mkl_lnx_openmp":  ["@mkl_dnn_v1//:mkl_dnn"],
+        "@org_tensorflow//third_party/mkl:build_with_mkl_windows_openmp": ["@mkl_dnn_v1//:mkl_dnn"],
         "//conditions:default": [],
     })
