diff --git a/tensorflow/lite/tools/evaluation/BUILD b/tensorflow/lite/tools/evaluation/BUILD
index 02ca40fad25cf..c2b12b7741dfc 100644
--- a/tensorflow/lite/tools/evaluation/BUILD
+++ b/tensorflow/lite/tools/evaluation/BUILD
@@ -54,7 +54,6 @@ cc_library(
         ],
         "//conditions:default": [],
     }) + select({
-        "//tensorflow:linux_armhf": [],
         "//tensorflow:linux_s390x": [],
         "//conditions:default": [
             "//tensorflow/lite/delegates/xnnpack:xnnpack_delegate",
